src/cdk/table/table.ts

```diff
@@ -375,16 +375,20 @@ export class CdkTable<T> implements AfterContentChecked, CollectionViewer, OnDes
    * The column definitions provided by the user that contain what the header, data, and footer
    * cells should render for each column.
    */
-  @ContentChildren(CdkColumnDef) _contentColumnDefs: QueryList<CdkColumnDef>;
+  @ContentChildren(CdkColumnDef, {descendants: true}) _contentColumnDefs: QueryList<CdkColumnDef>;
```

Comment:
Note that this makes assumptions that we can't nest tables. AFAIK it's not something we support anyway.

Reply:
Right, based on the trick we use the get cell outlets after stamping out row views, we absolutely, 100% do not support nested tables.